File: TI1/TI1.2 SOAP_DOTNET_SINBDD_GR09/01.SERVIDOR/CONUNI_SOAP_DOTNET_GR05/CONUNI_SOAP_DOTNET_GR09/AutenticacionService.cs
using System;

namespace CONUNI_SOAP_DOTNET_GR09
{
    /// <summary>
    /// Servicio de autenticación centralizado en el servidor.
    /// Las credenciales residen ÚNICAMENTE aquí, en el servidor.
    /// Los clientes deben invocar la operación Login del WebService.
    /// </summary>
    public class AutenticacionService
    {
        // Credenciales almacenadas SOLO en el servidor
        private const string UsuarioValido    = "MONSTER";
        private const string ContrasenaValida = "MONSTER9";

        /// <summary>
        /// Valida las credenciales de un usuario.
        /// </summary>
        /// <param name="usuario">Nombre de usuario enviado por el cliente</param>
        /// <param name="contrasena">Contraseña enviada por el cliente</param>
        /// <returns><c>true</c> si las credenciales son correctas</returns>
        public bool ValidarCredenciales(string usuario, string contrasena)
        {
            if (string.IsNullOrEmpty(usuario) || string.IsNullOrEmpty(contrasena))
                return false;

            return UsuarioValido.Equals(usuario.Trim().ToUpper())
                && ContrasenaValida.Equals(contrasena.Trim());
        }
    }
}

File: Proyectos/TI.1 PAO202650_SINBDD_NRC30742/TI.1.1. PAO202650_SOAP_JAVA_SINBDD_NRC30742/04. SERVIDOR/CONUNI_SOAP_JAVA_GR05/src/main/java/ec/edu/monster/servicios/AutenticacionService.java
package ec.edu.monster.servicios;

/**
 * Servicio de autenticación centralizado en el servidor.
 * Las credenciales residen ÚNICAMENTE aquí, en el servidor.
 * Los clientes deben llamar al WebService para validar credenciales.
 *
 * @author GR09 - Arquitectura de Software
 */
public class AutenticacionService {

    // Credenciales almacenadas SOLO en el servidor
    private static final String USUARIO_VALIDO    = "MONSTER";
    private static final String CONTRASENA_VALIDA = "MONSTER9";

    /**
     * Valida las credenciales de un usuario.
     *
     * @param usuario   nombre de usuario enviado por el cliente
     * @param contrasena contraseña enviada por el cliente
     * @return {@code true} si las credenciales son correctas, {@code false} en caso contrario
     */
    public boolean validarCredenciales(String usuario, String contrasena) {
        if (usuario == null || contrasena == null) {
            return false;
        }
        return USUARIO_VALIDO.equals(usuario.trim().toUpperCase())
                && CONTRASENA_VALIDA.equals(contrasena.trim());
    }
}
